build/lpc55pins/src/lib.rs

```diff
@@ -140,21 +207,27 @@ impl ToTokens for PinConfig {
 pub fn codegen(pins: Vec<PinConfig>) -> Result<()> {
     let out_dir = build_util::out_dir();
     let dest_path = out_dir.join("pin_config.rs");
-    let mut file = std::fs::File::create(dest_path)?;
-
+    let mut file = std::fs::File::create(&dest_path)?;
     let mut buf = BufWriter::new(Vec::new());
+    let mut used_slots = 0u8;
+
     if pins.iter().any(|p| p.name.is_some()) {
-        writeln!(&mut buf, "use drv_lpc55_gpio_api::Pin;")?;
+        writeln!(&mut file, "use drv_lpc55_gpio_api::Pin;")?;
```

Comment:
This doesn't look write, the goal was to write everything into the buf first and then write to the file at the end

Reply:
I'll revert this line.